addons/web/static/tests/widgets/company_switcher_tests.js

```diff
@@ -125,6 +127,33 @@ QUnit.module('widgets', {
             });
         });
 
+        QUnit.test("Toggle multiple companies in one go in multiple company mode", async function (assert) {
+            /**
+             *          [x] **Company 1**          [x] **Company 1**
+             *  toggle->[ ] Company 2     ====>    [x] Company 2
+             *  toggle->[x] Company 3              [x] Company 3
+             */
+            assert.expect(2);
+            SwitchCompanyMenu.prototype.TOGGLE_DELAY = 1000;
+
+            const menu = await createSwitchCompanyMenu({
+                session: {
+                    ...this.session_mock_multi,
+                    setCompanies: function (mainCompanyId, companyIds) {
+                        assert.equal(mainCompanyId, 1, "Main company should not have changed");
+                        assert.equal(companyIds.length, 2, "First and Second company should be activated");
```

Comment:
in the comment above, you say that you should have the 3 companies allowed at the end

Reply:
OK, I will change comment, as test has no issue, I forgot to change comment.